subscribe.go

```diff
@@ -169,21 +175,48 @@ func NewInMemSubscriber(messagesByTopic map[string][]SubMessage) *InMemSubscribe
 	for topic, msgs := range messagesByTopic {
 		byTopic[topic] = msgs
 	}
-	return &InMemSubscriber{
-		msgs: byTopic,
-	}
+	return &InMemSubscriber{msgs: byTopic}
+}
+
+// Topic returns named topic handle.
+// Seeded messages are copied for each topic handle
+func (s *InMemSubscriber) Topic(topic string) SubTopic {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	return NewInMemSubTopic(s.msgs[topic])
+}
+
+// Close forgets seeded messages.
+func (s *InMemSubscriber) Close() error {
+	s.mu.Lock()
+	s.msgs = nil
+	s.mu.Unlock()
+	return nil
+}
```

Comment:
@dim I did not bother with remembering topics in InMemSubscriber (to clean them up on close).

Btw, I'd prefer to have not in-mem mocks, but smth channel-backed:

```go
pub, sub := NewChannel( /* buffer size */ 0 )
_ = pub.Topic("foo").Publish(...)
_ = sub.Topic("foo").Subscribe(...)
```

This is nice to check background subscriptions (like, updates subscription, that can be ran in background by lib user).

Reply:
that for mem only?

Reply:
@dim I don't really get the question. Channel-based pub-sub - for testing, mem only - if you meant this.

Reply:
Maybe I'll get to it later ^^